Fix endpoint of string_split

yongtang opened the pull request:
It looks like at the moment, `string_split` is expose as `@tf_export("string_split")` which supports both v1 and v2.

And `string_split_v2` is exposed as `@tf_export("strings.split")` which support both v1 and v2 as well.

I think it makes sense to expose `@tf_export("strings.split")` in v1 and v2 for
`string_split_v2`.

But `@tf_export("string_split")` should really be v1 only.

This fix changes `@tf_export("string_split")` to `@tf_export(v1=["string_split"])`

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Fix endpoint of string_split

It looks like at the moment, `string_split` is expose as
`@tf_export("string_split")` which supports both v1 and v2.
And `string_split_v2` is exposed as `@tf_export("strings.split")`
which support both v1 and v2 as well.

I think it makes sense to expose `@tf_export("strings.split")` for
`string_split_v2`, but `@tf_export("string_split")` should really
be v1 only.

This fix changes `@tf_export("string_split")` to `@tf_export(v1=["string_split"])`

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
Also /cc @martinwicke

martinwicke commented on tensorflow/python/ops/string_ops.py:
```diff
@@ -192,7 +192,7 @@ def string_format(template, inputs, placeholder="{}", summarize=3, name=None):
 
 
-@tf_export("string_split")
+@tf_export(v1=["string_split"])
```
Yes, that's a mistake, thanks!

yongtang committed:
Update golden with

bazel-bin/tensorflow/tools/api/tests/api_compatibility_test
           --update_goldens True

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang committed:
Update tf_upgrade_v2.py to add string_split -> tf.compat.v1.string_split,

The behavior of `string_split` (v1) is different from `strings.split` (v2)

Signed-off-by: Yong Tang <yong.tang.github@outlook.com>

yongtang commented:
The PR has been updated to fix the api compatibility test. The tf_upgrade_v2.py has also been updated so that v1's `tf.string_split` points to v2's `tf.compat.v1.string_split` (the v2's `tf.strings.split` behaves differently) so rename is needed.

yongtang commented:
Also ping @rthadur, any update on this PR?

yongtang commented:
Ping @rthadur, just rebased to fix the merge conflict. Would hope this PR could land in 2.0 as it addresses the incorrect api endpoint issue. Otherwise it will be really hard to make a change after 2.0 is out.
